src/nupic/engine/Network.cpp

```diff
@@ -980,9 +980,14 @@ void Network::resetProfiling()
     regions_.getByIndex(i).second->resetProfiling();
 }
 
-void Network::registerRegionPackage(const char * path)
+void Network::registerPyRegionPackage(const char * path)
 {
-  Region::registerRegionPackage(path);
+  Region::registerPyRegionPackage(path);
+}
+
+void Network::registerCPPRegion(const std::string name, GenericRegisteredRegionImpl* wrapper)
```

Comment:
You should probably forward declare GenericRegisteredRegionImpl in this file. It compiles fine because of the declaration in the header but I think it is good practice to include in this file as well.